src/validation.cpp

```diff
@@ -1159,6 +1159,14 @@ CTransactionRef GetTransaction(const CBlockIndex* const block_index, const CTxMe
 {
     LOCK(cs_main);
 
+    if (mempool) {
+        CTransactionRef ptx = mempool->get(hash);
+        if (ptx) return ptx;
+    }
+    if (g_txindex) {
+        CTransactionRef tx;
+        if (g_txindex->FindTx(hash, hashBlock, tx)) return tx;
```

Comment:
Something like the following would keep behavior?
```cpp
uint256 block_hash;
if (g_txindex->FindTx(hash, block_hash, tx)) {
    if (block_index && block_index->GetBlockHash() != block_hash) return nullptr;
    hashBlock = block_hash;
    return tx;
}
```
Note the local `block_hash` to avoid changing the out-arg in case it fails.

Reply:
Makes sense; I can add a similar check for the mempool condition.